File: src/app/services/IPowerMeterService.ts
import { PowerMeterData } from "../model/PowerMeterData";
import { PowerMeterServiceEvent } from "../model/PowerMeterServiceEvent";

export interface IPowerMeterService {
    connect(): Promise<void>;
    disconnect(): Promise<void>;
    reconnectToLastConnected(): Promise<void>;
    selectNewDevice(): Promise<void>;
    subscribeForPowerData(callback: (event: PowerMeterServiceEvent) => void): void;
    unsubscribeForPowerData(callback: (event: PowerMeterServiceEvent) => void): void;
    readonly deviceSelected: boolean;
    readonly deviceName: string;
    readonly isConnected: boolean;
}

File: android/app/src/main/java/com/openbikecomputer/service/IPowerMeterService.java
package com.openbikecomputer.service;

/**
 * Bluetooth cycling power meter. Ported from {@code IPowerMeterService.ts}.
 */
public interface IPowerMeterService {
    void selectNewDevice();

    void reconnectToLastConnected();

    void disconnect();

    boolean isConnected();

    boolean isDeviceSelected();

    String getDeviceName();

    void subscribeForPowerData(OnPowerData listener);

    void unsubscribeForPowerData(OnPowerData listener);
}
